### ISSUE TITLE
User-Customizable Theme

### ISSUE THREAD

**Issue by lmparppei** [lmparppei/Beat#44]
Create a user-customizable theme both for light & dark mode.

The Theme system is a pretty convoluted one, but originally allowed multiple themes, so it should be easy enough to implement. The harder part is the UI. Beat promises it has no preferences, complicated menus and no nonsense, but a simple color customization menu shouldn't be too much. :-)

Beat uses a class called `DynamicColor` which has both the light and dark mode colors, and `ThemeManager` creates the dynamic colors according to `Themes.plist`. When creating a new theme, the user would *have* to determine colors for both dark & light mode to avoid confusion. You could have a checkbox to "use same color" or something to minimize hassle.

This should be quite easy to implement, and you can use Swift if you want to.

**Comment by lmparppei** [comment]
User-created themes would also require restructuring the plist file. Right now the light/dark are apparently separate themes, so the dictionary should probably be structured as such:
```
Theme: {
    "Background Light": rgb,
    "Background Dark": ...
}
```

Or, alternatively:
```
Theme: {
    "Light": { "Background": rgb },
    "Dark": { }
}
```

Latter would keep the color setting keys uniform, while the first one would probably be easier to handle in many cases.

We can't save the user-created theme in the bundled plist, so it would have to be saved in the app support folder or in user preferences.

**Comment by lmparppei** [comment]
![image](https://user-images.githubusercontent.com/20226704/105036471-30b07700-5a65-11eb-9638-36eeb3478c28.png)

Color theme can be customized now. It doesn't save the color scheme for now, and  I have to figure out what would be the best way of doing this.

**Comment by lmparppei** [comment]
For now, I will only allow the user to customize and reset the colors. If somebody has the willpower to create some sort of theme browser and support, I can consider adding it to the app. Still, I want Beat to be as fail-proof as possible, and too many moving parts will only make it harder to optimize and maintain.

**Comment by lmparppei** [comment]
First iteration is now implemented and will be released in 1.7b.

**Comment by harp37** [comment]
Dope, look forward to checking it out!
